presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/TestHiveHadoop2Plugin.java

```diff
@@ -99,6 +99,24 @@ public void testGcsAccessTokenAndHiveCachingMutuallyExclusive()
                 .hasMessageContaining("Use of GCS access token is not compatible with Hive caching");
     }
 
+    @Test
+    public void testImmutablePartitionsAndInsertOverwriteMutuallyExclusive()
+    {
+        Plugin plugin = new HiveHadoop2Plugin();
+        ConnectorFactory connectorFactory = Iterables.getOnlyElement(plugin.getConnectorFactories());
+
+        assertThatThrownBy(() -> connectorFactory.create(
+                "test",
+                ImmutableMap.<String, String>builder()
+                        .put("hive.insert-existing-partitions-behavior", "APPEND")
+                        .put("hive.immutable-partitions", "true")
+                        .put("hive.metastore.uri", "thrift://foo:1234")
+                        .build(),
+                new TestingConnectorContext())
+                .shutdown())
```

Comment:
shutdown won't be called anyway.

Reply:
Thank you for the info. I'll left it as is to be consistent with other places.

Reply:
Please don't. It may suggest that exception is thrown by `shutdown` method.

Reply:
done.